File: cs-499 project java/Task/TaskService.java
/*
 * Name: Justin Fifarek
 * Course: SNHU CS-320
 * Date: 11/24/2023
 * 
 * Module 4-1 Milestone - Task service class
 */

package Task;
import java.util.ArrayList;

public class TaskService {
	// Array list to hold tasks
	public ArrayList<Task> taskList = new ArrayList<Task>();
	
	// Display tasks
	public void displayTasks() {
		for (int i = 0; i < taskList.size(); i++) {
			System.out.println("Task ID:  " + taskList.get(i).getTaskID());
			System.out.println("Task Name: " + taskList.get(i).getTaskName());
			System.out.println("Task Description: " + taskList.get(i).getTaskName());
		}
	}
	
	// Returns task given a taskID
	// Returns default task if no matching taskID is found
	public Task getTask(String taskID) {
		Task task = new Task(null, null);
		for (int i = 0; i < taskList.size(); i++) {
			if (taskList.get(i).getTaskID().contentEquals(taskID)) {
				task = taskList.get(i);
			}
		}
		return task;
	}
	
	// Adds new task
	public void addTask(String taskName, String taskDesc) {
		Task task = new Task(taskName, taskDesc);
		taskList.add(task);
	}
	
	// Delete task given a taskID
	public void deleteTask(String taskID) {
		for (int i = 0; i < taskList.size(); i++) {
			if (taskList.get(i).getTaskID().equals(taskID)) {
				taskList.remove(i);
				break;
			}
			if (i == taskList.size() - 1) {
				System.out.println("Task ID \'" + taskID + "' not found.");
			}
		}
	}
	
	// Update given taskID's taskName with given string
	public void updateTaskName(String newTaskName, String taskID) {
		for (int i = 0; i < taskList.size(); i++) {
			if (taskList.get(i).getTaskID().equals(taskID)) {
				taskList.get(i).setTaskName(newTaskName);
				break;
			}
			if (i == taskList.size() - 1) {
				System.out.println("Task ID: " + taskID + " not found.");
			}
		}
	}
	
	// Update given taskID's taskDesc with given string
	public void updateTaskDesc(String newTaskDesc, String taskID) {
		for (int i = 0; i < taskList.size(); i++) {
			if (taskList.get(i).getTaskID().equals(taskID)) {
				taskList.get(i).setTaskDesc(newTaskDesc);
				break;
			}
			if (i == taskList.size() - 1) {
				System.out.println("Task ID: " + taskID + " not found.");
			}
		}
	}
}

File: docs/module3/apt-management-app/Task/TaskService.cpp
/*
    Name: Justin Fifarek
    Project: TaskService class
*/
#include <iostream>
#include <vector>
#include "Task.cpp"

class TaskService {
private:
    // Vector to hold tasks
    std::vector<Task> taskList;

public:
    // Display tasks
    void displayTasks() {
        for (size_t i = 0; i < taskList.size(); ++i) {
            std::cout << "Task ID:  " << taskList[i].getTaskID() << std::endl;
            std::cout << "Task Name: " << taskList[i].getTaskName() << std::endl;
            std::cout << "Task Description: " << taskList[i].getTaskDesc() << std::endl;
        }
    }

    // Searches through taskList vector and returns the task
    // of given taskID, if it exists.
    Task getTask(std::string taskID) {
        for (size_t i = 0; i < taskList.size(); ++i) {
            if (taskList[i].getTaskID() == taskID) {
                return taskList[i];
            }
        }
        return Task("", "");
    }

    // Adds new task using the task constructor
    void addTask(std::string taskName, std::string taskDesc) {
        Task task(taskName, taskDesc);
        taskList.push_back(task);
    }

    // Delete task given a taskID
    void deleteTask(const std::string& taskID) {
        for (auto it = taskList.begin(); it != taskList.end(); ++it) {
            if (it->getTaskID() == taskID) {
                taskList.erase(it);
                return;
            }
        }
        std::cout << "Task ID " << taskID << " not found." << std::endl;
    }

    // Searches through taskList vector and updates taskName
    // of given taskID, if it exists.
    void updateTaskName(const std::string& newTaskName, const std::string& taskID) {
        for (auto& task : taskList) {
            if (task.getTaskID() == taskID) {
                task.setTaskName(newTaskName);
                return;
            }
        }
        std::cout << "Task ID: " << taskID << " not found." << std::endl;
    }

    // Searches through taskList vector and updates taskDesc
    // of given taskID, if it exists.
    void updateTaskDesc(const std::string& newTaskDesc, const std::string& taskID) {
        for (auto& task : taskList) {
            if (task.getTaskID() == taskID) {
                task.setTaskDesc(newTaskDesc);
                return;
            }
        }
        std::cout << "Task ID: " << taskID << " not found." << std::endl;
    }
};
